src/core/dbus-service.c

```diff
@@ -294,6 +294,29 @@ int bus_service_method_dump_file_descriptor_store(sd_bus_message *message, void
         return sd_bus_send(NULL, reply, NULL);
 }
 
+#if __SIZEOF_SIZE_T__ == 8
+static int property_get_size_as_uint32(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        size_t *value = ASSERT_PTR(userdata);
+        uint32_t sz = *value >= UINT32_MAX ? UINT32_MAX : (uint32_t) *value;
+
+        /* Returns a size_t as a D-Bus "u" type, i.e. as 32bit value, even if size_t is 64bit. We'll saturate if it doesn't fit. */
+
+        return sd_bus_message_append_basic(reply, 'u', &sz);
+}
+#elif __SIZEOF_SIZE_T__ == 4
+#define property_get_size_as_uint32 NULL
```

Comment:
why NULL instead of bus_property_get_unsigned?

Reply:
dbus' "u" type is an uint32_t. "unsigned" might be anything really. at least in theory.

Here we validated that the size of size_t is 4, hence we want to use uint32_t. And if no property getty is set, then sd-bus automatically reads uint32_t for "u". Hence NULL.

(note that bus_property_get_unsigned() actually also maps to NULL, but that's "secret". ;-) i.e. on all archs we support unsigned is actually 32bit...)

Reply:
> And if no property getty is set, then sd-bus automatically reads uint32_t for "u". Hence NULL.

Ah, I was missing this fallback